lib/bolt/transport/winrm.rb

```diff
@@ -100,8 +100,19 @@ def run_script(target, script, arguments, _options = {})
       end
 
       def run_task(target, task, arguments, _options = {})
-        executable = target.select_impl(task, PROVIDED_FEATURES)
-        raise "No suitable implementation of #{task.name} for #{target.name}" unless executable
+        if from_api?(task)
+          # TODO: Remove as part of BOLT-664
+          Dir.mktmpdir do |dir|
```

Comment:
The tmp directory will be deleted after exiting the block. That doesn't seem like it'll work. I think you want to use the non-block variant (then delete it at the end).